### PR TITLE
feat(linux): Implement Options page and option to disable Sentry error reporting

### PR DESCRIPTION
**Pull request by ermshiperete**
This change displays tabs at the left and adds an "Options" page. This makes it look closer to the Windows version.
It also implements the option to disable Sentry error reporting (#5027).

New look:
image

Previously:
image

## The new Options tab:

Screenshot from 2023-05-26 17-45-59

# User Testing

## Preparations

- The tests can be run on any Linux platforms.
- [Install build artifacts of this PR](https://github.com/keymanapp/keyman/wiki/How-to-test-artifacts-from-pull-requests-for-Keyman-for-Linux)
- Reboot
- In a terminal window, run `unset KEYMAN_NOSENTRY`
- start the following tests from this terminal window!

## Tests

**TEST_UI_DEFAULT:**

- Start UI with `km-config`
- switch to `Options` tab
- verify that "Automatically report errors to keyman.com" is checked
- in the terminal run `gsettings get com.keyman.options error-reporting` and verify that the output is `true`

**TEST_UI_DEFAULT_PERSIST:**

- continuing from the previous test, close Keyman Configuration.
- restart it from the terminal by running `km-config`
- verify that "Automatically report errors to keyman.com" is checked

**TEST_UI_DISABLE:**

- continuing from the previous test
- uncheck "Automatically report errors to keyman.com"
- in the terminal run `gsettings get com.keyman.options error-reporting` and verify that the output is `false`

**TEST_UI_DISABLE_PERSIST:**

- continuing from the previous test, close Keyman Configuration.
- restart it from the terminal by running `km-config`
- verify that "Automatically report errors to keyman.com" is unchecked

**TEST_SETTINGS_ENABLED:**

- continuing from the previous test
- in the terminal, run `gsettings set com.keyman.options error-reporting true`
- verify that "Automatically report errors to keyman.com" on the Options tab is checked

**TEST_SETTINGS_ENABLED2:**

- continuing from the previous test
- in the terminal, run `gsettings set com.keyman.options error-reporting true`
- verify that "Automatically report errors to keyman.com" on the Options tab is still checked

**TEST_SETTINGS_DISABLED:**

- continuing from the previous test
- in the terminal, run `gsettings set com.keyman.options error-reporting false`
- verify that "Automatically report errors to keyman.com" on the Options tab is unchecked

**TEST_ENV_VAR_SET:**

- close Keyman Configuration
- in the terminal run `export KEYMAN_NOSENTRY=1`
- run `km-config`
- on the Options tab, verify that "Automatically report errors to keyman.com" is grayed out and can't be toggled

### CODE REVIEW

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_UI_DISABLE_PERSIST (PASSED):** Verified "Automatically report errors to keyman.com" is unchecked.

   

* **TEST_SETTINGS_ENABLED (PASSED):** Verified "Automatically report errors to keyman.com" on the Options tab is checked. 

   

* **TEST_SETTINGS_ENABLED2 (PASSED):** Verified that  "Automatically report errors to keyman.com" on the Options tab is still checked.


* **TEST_SETTINGS_DISABLED (PASSED):** Verified that "Automatically report errors to keyman.com" on the Options tab is unchecked.

   

* **TEST_ENV_VAR_SET (PASSED):** Verified that "Automatically report errors to keyman.com" on the Options tab is grayed out and can't be toggled.

File: linux/keyman-config/com.keyman.gschema.xml [hunk 1]
```diff
@@ -1,8 +1,12 @@
 <schemalist>
   <schema id="com.keyman.options" path="/desktop/ibus/keyman/options/">
-
     <child name="keyboard-options" schema="com.keyman.options.child"/>
 
+    <key name="error-reporting" type="b">
+        <default>true</default>
+        <summary>Automatic error reporting</summary>
+        <description>true to enable automatic reporting of errors to Sentry</description>
```
Comment (mcdurdin) [inline linux/keyman-config/com.keyman.gschema.xml:8]: Will these be localizable?

**Review by mcdurdin — APPROVED** [review]
LGTM

File: linux/keyman-config/com.keyman.gschema.xml (hunk 1 above)
Comment (ermshiperete) [reply linux/keyman-config/com.keyman.gschema.xml:8]: Currently not. [It seems to be possible to add localization for schemas](https://docs.gtk.org/gio/class.Settings.html), but I can't find details how to do that. And I think the important part is that it's localized in the UI - if a user wants to deal directly with the settings I assume he has some knowledge of English.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.115-alpha](https://keyman.com/downloads/releases/alpha/17.0.115)

File: linux/keyman-config/com.keyman.gschema.xml (hunk 1 above)
Comment (mcdurdin) [reply linux/keyman-config/com.keyman.gschema.xml:8]: Righto -- I missed that the UI is localizable already and these were internal strings. All good :)

**Pull request merged**